hypersync-client/src/from_arrow.rs

```diff
@@ -11,49 +12,105 @@ use crate::{
 /// Used to do ArrowBatch-Native Rust type conversions while consuming the input value.
 pub trait FromArrow: Sized {
     /// Converts to the Vector type from the ArrowBatch type.
-    fn from_arrow(batch: &ArrowBatch) -> Vec<Self>;
+    /// B is the type of binary array used to downcast since we use both binary views and
+    /// binary arrays
+    fn from_arrow_bin_array<B: List<Item = [u8]> + 'static, U: List<Item = str> + 'static>(
+        batch: &ArrowBatch,
+    ) -> Vec<Self>;
+
+    /// Default implementation that uses the binary array type.
+    /// IPC is written with regular binary arrays.
+    /// Named 'from_arrow' for backwards compatibility.
+    fn from_arrow(batch: &ArrowBatch) -> Vec<Self> {
+        Self::from_arrow_bin_array::<BinaryArray<i32>, Utf8Array<i32>>(batch)
+    }
+
+    /// An additional method that uses the binary view array type.
+    /// This is to be able to reuse the trait server side where Binary Views are
+    /// used instead.
+    fn from_arrow_bin_view_array(batch: &ArrowBatch) -> Vec<Self> {
+        Self::from_arrow_bin_array::<BinaryViewArray, Utf8ViewArray>(batch)
+    }
```

Comment:
So now you can see I've made the two different functions for binary view arrays and the simple binary arrays passing in the actual types, which will downcast the arrow to the right types. And then I've made from_arrow the same as before. So it's backwards compatible. Nothing else breaks. I could have, for example, just implemented the list traits inside of Scar and for the binary view arrays and then just used this API there. But I just thought it's easier to put it inside the client so it's obvious why I did it this way.